apps/condo/domains/marketplace/access/InvoiceContext.js

```diff
@@ -12,13 +12,19 @@ const {
     queryOrganizationEmployeeFromRelatedOrganizationFor,
     checkPermissionInUserOrganizationOrRelatedOrganization,
 } = require('@condo/domains/organization/utils/accessSchema')
-const { RESIDENT } = require('@condo/domains/user/constants/common')
+const { RESIDENT, SERVICE } = require('@condo/domains/user/constants/common')
+const { canDirectlyManageSchemaObjects, canDirectlyReadSchemaObjects } = require('@condo/domains/user/utils/directAccess')
 
-async function canReadInvoiceContexts ({ authentication: { item: user } }) {
+
+async function canReadInvoiceContexts ({ authentication: { item: user }, listKey }) {
     if (!user) return throwAuthenticationError()
     if (user.deletedAt) return false
     if (user.isAdmin || user.isSupport) return {}
 
+    if (user.type === SERVICE) {
+        return await canDirectlyReadSchemaObjects(user, listKey)
+    }
```

Comment:
I don’t think it’s worth adding restrictions by user type for this check

These are accesses like support, only with flexible settings

```suggestion
    const hasDirectAccess = await canDirectlyReadSchemaObjects(user, listKey)
    if (hasDirectAccess) return true
```